src/utils/enrichment.ts

```diff
@@ -0,0 +1,47 @@
+/*
+	Copyright 2020 Dynatrace LLC
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+import { readFileSync } from "fs";
+import { Dimension } from "./metric/metric";
+
+const cIndirectionFilename = "dt_metadata_e617c525669e072eebe3d0f08212e8f2.json";
+
+function readOneAgentMetadata(filename: string): Record<string, string> {
+	const indirectionFileContents = readFileSync(filename).toString("utf-8").trim();
+	if (indirectionFileContents === "") {
```

Comment:
Will this return `""` if the file does not exist or if the process is lacking the permissions to open it?

Reply:
Neither. I noticed the test in Georg's projects and decided to cover the case just to be safe.

Reply:
Ah I just noticed it will throw and that this will be caught further up in the the stack. Fine then.